cfme/tests/cloud/test_providers.py

```diff
@@ -1328,7 +1328,23 @@ def test_ec2_regions_up_to_date():
         expectedResults:
             1. There should be same regions in CFME as in AWS Console.
     """
-    pass
+    regions = provider.mgmt.list_regions(verbose=True)
+    view = navigate_to(CloudProvider, 'Add')
+    view.prov_type.fill("Amazon EC2")
+    all_options = view.region.all_options
+    # Delete option <Choose>
+    all_options.pop(0)
+    all_options_texts = [option.text for option in all_options]
+    regions_not_in_cfme = []
+    for region in regions:
+        if region not in all_options_texts:
+            regions_not_in_cfme.append(region)
+        else:
+            all_options_texts.remove(region)
+    if len(regions_not_in_cfme) > 0:
+        pytest.fail("Regions {} are not in CFME!".format(regions_not_in_cfme))
+    if len(all_options_texts) > 0:
+        pytest.fail("Extra regions in CFME: {}".format(all_options_texts))
```

Comment:
Could you do something like:
```suggestion
    assert sorted(regions_provider) == sorted(regions_cfme)
```
And then log the two lists so that we know what regions are missing if the test fails.

Reply:
Okay, this looks good, but I would like to list only regions that are extra or missing in CFME so we don't have to manually compare these two lists.

Reply:
Can you do something like https://stackoverflow.com/questions/3462143/get-difference-between-two-lists?

Reply:
Okay, I used set - set to get diff of these two lists.